app/code/core/Mage/Core/Model/Encryption.php

```diff
@@ -153,53 +148,53 @@ public function validateHashByVersion(#[\SensitiveParameter] $password, $hash, $
         return hash_equals($this->hash($salt . $password, $version), $hash);
     }
 
-    /**
-     * Instantiate crypt model
-     *
-     * @param string $key
-     * @return Varien_Crypt_Mcrypt
-     */
-    protected function _getCrypt($key = null)
+    public function encrypt(#[\SensitiveParameter] string $data): string
     {
-        if (!$this->_crypt) {
-            if ($key === null) {
-                $key = (string) Mage::getConfig()->getNode('global/crypt/key');
-            }
-            $this->_crypt = Varien_Crypt::factory()->init($key);
-        }
-        return $this->_crypt;
+        $key = Mage::getEncryptionKeyAsBinary();
+        $nonce = random_bytes(SODIUM_CRYPTO_SECRETBOX_NONCEBYTES);
+        $ciphertext = sodium_crypto_secretbox($data, $nonce, $key);
+        $encrypted = sodium_bin2base64($nonce . $ciphertext, SODIUM_BASE64_VARIANT_ORIGINAL);
+        sodium_memzero($data); // Clean sensitive data from memory
```

Comment:
```suggestion
        // Clean sensitive data from memory
        sodium_memzero($data);
        sodium_memzero($key);
        sodium_memzero($nonce);
        sodium_memzero($ciphertext);
```

Should we zero out these values too? The example from the link below seems to do so:

https://davegebler.com/post/php/php-encryption-the-right-way-with-libsodium